source/compiler/qsc_eval/src/lib.rs

```diff
@@ -1776,6 +1782,11 @@ fn eval_binop_eq(lhs_val: Value, rhs_val: Value, rhs_span: PackageSpan) -> Resul
             // to prevent executing programs that do result comparisons.
             Err(Error::ResultComparisonUnsupported(rhs_span))
         }
+        (Value::Result(val::Result::Loss), _) | (_, Value::Result(val::Result::Loss)) => {
```

Comment:
So basically loss != loss (and causes a runtime error). Similar to a NAN handling. l like this.